File: my-code/810/main.py
# -*- coding:utf-8 -*-
# @Time     : 2021/5/25 07:27
# @Author   : Ranshi
# @File     : 810. 黑板异或游戏.py
from functools import reduce
from operator import xor
from typing import List


class Solution:
    def xorGame(self, nums: List[int]) -> bool:
        if len(nums) % 2 == 0:
            return True
        xor_sum = reduce(xor, nums)
        return xor_sum == 0


if __name__ == '__main__':
    s = Solution()
    print(s.xorGame(nums=[1, 1, 2]))

File: my-code/810/main.js
// @Time     : 2021/05/28 17: 47
// @Author   : Ranshi
// @File     : 810. 黑板异或游戏.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
const xorGame = function (nums) {
    if (nums.length % 2 === 0) {
        return true
    }
    let xor = 0
    for (const count of nums) {
        xor ^= count
    }
    return xor === 0
};

console.log(xorGame(nums = [1, 1, 2]))